firmware/application/ui_navigation.cpp

```diff
@@ -555,6 +560,76 @@ void SystemStatusView::rtc_battery_workaround() {
     }
 }
 
+void SystemStatusView::new_sdcard_structure_checker() {
+    const std::filesystem::path root_dir = u"/";
+    const std::filesystem::path resources_dir = u"SYS";
+    std::vector<std::filesystem::path> directories = scan_root_directories(root_dir);
+
+    auto scan_result = std::find(directories.begin(), directories.end(), resources_dir);
+
+    auto sd_status = sd_card::status();
+
+    if (
+        nav_.is_valid() &&
+        (scan_result == directories.end()) &&
+        (sd_status == sd_card::Status::Mounted)) {
+        nav_.display_modal(
+            "Warning",
+            "        WARNING!!!\n"
+            "SD Card content changed\n"
+            "in this update,\n"
+            "Press YES to auto move.\n"
+            "After finished,\n"
+            "you should move your own\n"
+            "files back into\n"
+            "original folders,\n"
+            "which are now only for users.\n"
+            "You can also choose NO,\n"
+            "and then manually merge\n"
+            "following the Wiki.\n"
+            "Details:\n"
+            "Mayhem wiki - Updating",
+            COMPACTYESNO,
+            [this](bool choice) {
+                if (choice) {
+                    new_sdcard_structure_worker();
+                }
+            });
+    }
+}
+
+void SystemStatusView::new_sdcard_structure_worker() {
+    const std::filesystem::path root_dir = u"/";
+    const std::filesystem::path system_dir = u"SYS";
+
+    std::vector<std::filesystem::path> ignore_dirs = {u"FIRMWARE",
+                                                      u"APPS",
+                                                      u"SYS",
+                                                      u"SETTINGS",
+                                                      u"DEBUG",
+                                                      u"LOGS",
```

Comment:
I think distinguish between original FREQMAN files and user-modified FREQMAN files by looking at the file timestamps is not quite reliable, since you don’t know which version users are using. I can try not move FREQMAN tho.

Reply:
all fixed